### DISCUSSION TITLE
[SUGGESTION] Solution for uBlock not working on Chrome.

### DISCUSSION THREAD

**Discussion by alizaincodes** [uBlockOrigin/uBlock-issues#3416] [Ideas] [answered]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

# Solution

Hi there!
I have a suggestion. I have seen that uBlock is now not supported on Chrome.
I found a solution myself.

The extension uses network filters to block the ads.I know that Chrome has removed manifest v2 and switched to manifest v3.
And it provides poor support for networking features, the extension uses.

So instead of hanging around with the Chrome Manifest, you should start a local proxy server in the computer itself. And then set Chrome to pass through the proxy server.
In the proxy server, you should integrate the functionality of extension such as network filters.
This will be undetectable for Chrome and you can modify the body, headers and whatever easily.

Is this solution helpful?

### A specific URL where the issue occurs.

```yaml
It's a suggestion. I am not trying to submit a bug report.
```

### Steps to Reproduce

It's a suggestion.

### Expected behavior

It's a suggestion.

### Actual behavior

It's a suggestion.

### uBO version

latest

### Browser name and version

Chrome 128.0.6613.137

### Operating System and version

* This implies with every OS.

**Comment by garry-ut99** [comment]
> It's a suggestion. I am not trying to submit a bug report.

Then better place is https://github.com/uBlockOrigin/uBlock-issues/discussions.

> I have seen that uBlock is now not supported on Chrome. I found a solution myself.

Another solution is to switch to Chromium forks, which still support uBO, like mentioned there: 
- https://github.com/uBlockOrigin/uBlock-issues/wiki/About-Google-Chrome's-%22This-extension-may-soon-no-longer-be-supported%22#will-development-of-ubo-continue

> The extension uses network filters to block the ads.
> and you can modify the body, headers and whatever easily.

uBO uses far more than networks filters, see: 
- https://github.com/uBlockOrigin/uBlock-issues/issues/3309#issuecomment-2308747116 - at least points 1,2 and 5 still apply to proxy idea, proxy idea is better than hosts idea, but still not sufficient.

Besides, if proxy idea means that uBO would have to become an external application, then it was already discussed and declined here:
- https://github.com/uBlockOrigin/uBlock-issues/discussions/3023

**Reply by alizaincodes** [reply]
As you can see that Google Chrome has 65.72% of market share in browsers.
Millions of users use it daily. And the extension has also 1 Million users.

Why we cannot make an external application?
It's just a proxy server which extension's logic implemented in it.

The server will run locally on any specific port and then user can set the proxy settings in the Chrome settings.

**Reply by gothic-bum** [reply]
Read comment by gorhill (the developer of uBO): https://github.com/uBlockOrigin/uBlock-issues/discussions/3023#discussioncomment-7867379

**Reply by gorhill** [reply, marked as answer]
> It's just a proxy server which extension's logic implemented in it.

A suggestion: Can't you just do it?

**Reply by alizaincodes** [reply]
Okay :+1:

**Comment by Yuki2718** [comment]
Duplicate of https://github.com/uBlockOrigin/uBlock-issues/discussions/3023
